Source/Core/Core/Core.cpp

```diff
@@ -244,6 +251,10 @@ bool Init()
 	// Start the emu thread
 	s_emu_thread = std::thread(EmuThread);
 
+#ifdef USE_MEMORYWATCHER
+	s_memory_watcher = new MemoryWatcher;
```

Comment:
I think this line should be moved to https://github.com/spxtr/dolphin/blob/reporter/Source/Core/Core/Core.cpp#L366 , so you might skip the "Memory::IsInitialized()" check.